packages/dev/core/src/Engines/thinEngine.ts

```diff
@@ -1168,6 +1167,36 @@ export class ThinEngine extends AbstractEngine {
         this._bindUnboundFramebuffer(null);
     }
 
+    /**
+     * Generates mipmaps for the texture of the (single) render target
+     * @param texture The render target containing the texture to generate the mipmaps for
+     */
+    public generateMipMapsFramebuffer(texture: RenderTargetWrapper): void {
+        if (!texture.isMulti && texture.texture?.generateMipMaps && !texture.isCube) {
+            this.generateMipmaps(texture.texture);
+        }
+    }
+
+    /**
+     * Resolves the MSAA texture of the (single) render target into its non-MSAA version.
+     * Note that if "texture" is not a MSAA render target, no resolve is performed.
+     * @param texture  The render target texture containing the MSAA textures to resolve
+     */
+    public resolveFramebuffer(texture: RenderTargetWrapper): void {
+        const rtWrapper = texture as WebGLRenderTargetWrapper;
+        const gl = this._gl;
+
+        if (!rtWrapper._MSAAFramebuffer || rtWrapper.isMulti) {
+            return;
+        }
+
+        const bufferBits = rtWrapper._generateDepthBuffer ? gl.COLOR_BUFFER_BIT | gl.DEPTH_BUFFER_BIT : gl.COLOR_BUFFER_BIT;

```

Comment:
Should stencil be considered here? I would expect it to be blitted as well, but I'm not sure if it needs considerations elsewhere. 
In the project I work in, we utilize stencil when doing some postprocessing, and then it would be helpful if we can set up the PP target's FBO to share the depth-stencil texture with the resolved "main" render target's.

Reply:
Good call, I added code to resolve stencil too.

Reply:
I added switches to selectively enable resolving color, depth and stencil textures. To avoid breaking changes, resolving color textures is enabled by default, but not depth/stencil textures.